Automattic/jetpack
# Migrate to new Web fonts API format from Gutenberg

Fonts API is going through changes in Gutenberg v14.9 (planned, might still change as of Dec 27).

Resolves https://github.com/Automattic/jetpack/issues/28063

#### Changes proposed in this Pull Request:

* Support old and new array format passed to `wp_register_webfonts`
* Supports using the new utility class instead of deprecated utility functions
* Uses `class_exists( 'WP_Fonts' )` to see if we should support new version, or older version.
* **TODO**: rename some classes/functions to new format introduced in https://github.com/WordPress/gutenberg/pull/46749

#### Other information:


#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Build Gutenberg trunk, or use v14.9 if it has been released
* Ensure Jetpack Fonts module is enabled
* Test in site editor, open global styles panel, change fonts for headings/text etc and ensure fonts load and appear in the list
* Ensure fonts work at the frontend of the site
* Beware that Blockbase theme (and child themes) disable Jetpack fonts module, so use some other FSE theme like TT2/3 to test this.
